app/views/posts/show.html.erb

```diff
@@ -1,25 +1,37 @@
 <a href="<%= user_posts_path(@user) %>">
   <button class="custom-back-btn">⬅</button>
 </a>
+
 <div class="custom-post-container">
   <h2>Post <%= "#" + @post.id.to_s %> by <%= @user.name %></h2>
   <div class='custom-post-interactions'>
     <span>Comments: <%= @post.commentsCounter %>,</span>
-    <span>Likes: <%= @post.likesCounter  %></span>
+    <span>Likes: <%= @post.likesCounter %></span>
   </div>
   <p><%= @post.text %></p>
+  <div class='custom-like-section'>
+    <% if current_user.liked?(@post) %>
+      <%= button_to "Unlike", user_post_unlike_path(@user, @post), method: :delete, remote: true %>
+    <% else %>
+      <%= button_to "Like", user_post_like_path(@user, @post), method: :post, remote: true %>
+    <% end %>
+  </div>
 </div>
+
 <div class="custom-post-container">
-<h3>Comments:</h3>
-<% if @post.comments.present? %>
-  
-  <% @post.comments.each do |comment| %>
-    <div class="custom-post-comments">
-      <span><%= comment.user.name %>:</span>
-      <span><%= comment.text %></span>
-    </div>
+  <h3>Comments:</h3>
+  <% if @post.comments.present? %>
+    <% @post.comments.each do |comment| %>
+      <div class="custom-post-comments">
+        <span><%= comment.user.name %>:</span>
+        <span><%= comment.text %></span>
+      </div>
+    <% end %>
+  <% else %>
+    <p>No comments yet.</p>
   <% end %>
-<% else %>
-  <p>No comments yet.</p>
-<% end %>
-</div>
\ No newline at end of file
+</div>
+
+<a href="<%= new_user_post_comment_path(@user, @post) %>">
+  <button id="back-btn">Add a comment</button>
+</a>
```

Comment:
- [ ] Although post can be created successfully for given user, I'm unable to like nor comment on a post. When I click on a view post so as to be directed to the page where I can like or comment on the post, an error is thrown like below. It appears a bug was introduced in your code in an attempt to resolve the previous issue. Kindly ensure to fix the same.

![image](https://github.com/leloufadel/my_blog_app/assets/12745474/4f5889a2-4c8f-4011-9f4b-60916fd8be71)

![image](https://github.com/leloufadel/my_blog_app/assets/12745474/aeed8946-3d56-4409-bcad-87fdefc0becf)

Reply:
- [ ] Good Job @leloufadel  Fixed as shown below  screenshot 
![test](https://github.com/leloufadel/my_blog_app/assets/93598189/2e69a2d2-de82-4d58-8904-c6ef2d181f69)